backend-es/test/snapshots-out/Snapshot.RecursionSchemes01.js

```diff
@@ -13,22 +13,18 @@ const functorExprF = {
     $runtime.fail();
   }
 };
-const test1 = /* #__PURE__ */ (() => {
-  const go = v => {
-    if (v.tag === "Add") { return go(v._1) + go(v._2) | 0; }
-    if (v.tag === "Mul") { return go(v._1) * go(v._2) | 0; }
-    if (v.tag === "Lit") { return v._1; }
-    $runtime.fail();
-  };
-  return go;
-})();
-const test2 = /* #__PURE__ */ (() => {
-  const go = v => {
-    if (v.tag === "Add") { return go(v._1) + go(v._2) | 0; }
-    if (v.tag === "Mul") { return go(v._1) * go(v._2) | 0; }
-    if (v.tag === "Lit") { return v._1 + 1 | 0; }
-    $runtime.fail();
-  };
-  return go;
-})();
-export {$ExprF, Add, Lit, Mul, functorExprF, test1, test2};
+const test1$0$go$$rec = v => {
+  if (v.tag === "Add") { return test1$0$go$$rec(v._1) + test1$0$go$$rec(v._2) | 0; }
+  if (v.tag === "Mul") { return test1$0$go$$rec(v._1) * test1$0$go$$rec(v._2) | 0; }
+  if (v.tag === "Lit") { return v._1; }
+  $runtime.fail();
+};
+const test1 = test1$0$go$$rec;
```

Comment:
I think we should optimize the case of:
```
foo = let go = ... in go
```
For `LetRec`s to avoid generating an unnecessary bindings. This happens in a couple of the places in the snapshots.

Reply:
Yeah, I spotted that as well. The easiest thing would be, when we found a leading `LetRec` with a single term & where the body is just a local referring to the term, to get rid of the `LetRec` & change all of the locals to point to the `Var` of the expression. But I'm not sure if that's too specific - would that leave out any other cases that'd need to be optimized?